lib/volta.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import os
+import platform
+from shutil import which
+
+from devenv.constants import root
+from devenv.lib import archive
+from devenv.lib import fs
+from devenv.lib import proc
+
+VOLTA_HOME = f"{root}/volta"
+
+_version = "1.1.1"
+_sha256 = {
+    f"volta-{_version}-macos-aarch64.tar.gz": "013d419550525fa6a212c2693798f9e2e65737e887e3604b08bc15a6be737e01",  # noqa: E501
+    f"volta-{_version}-macos.tar.gz": "778ccaa943de8729d91e9df02a2767b470d97e8d5551faf6d89978db6f5f3c64",  # noqa: E501
+}
+
+
+def install() -> None:
+    unpack_into = f"{root}/bin"
+
+    if which("volta") == f"{unpack_into}/volta":
+        return
+
+    suffix = "-aarch64" if platform.machine() == "arm64" else ""
+    name = f"volta-{_version}-macos{suffix}.tar.gz"
+    url = "https://github.com/volta-cli/volta/releases/download/" f"v{_version}/{name}"
+
+    archive_file = archive.download(url, _sha256[name])
+    archive.unpack(archive_file, unpack_into)
```

Comment:
Isn't `subprocess.check_call('brew install --cask volta')` equivalent (but better maintained)?
Note the similarities (and differences):
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/v/volta.rb

Reply:
that cask only began existence this May so i'm wary of maintainership https://github.com/Homebrew/homebrew-cask/commits/aa461148bbb5119af26b82cccf5003e2b4e50d95/Casks/volta.rb?browsing_rename_history=true&new_path=Casks/v/volta.rb&original_branch=master

i would prefer to have control over how it's installed and it's pretty easy to do (as you can see)